src/services/jgroups/JGroupsService.ts

```diff
@@ -41,84 +49,113 @@ export class JGroupsService {
+            if (isNullOrUndefined(stackPromises[serverGroup])) {
+              stackPromises[serverGroup] = this.getDefaultStack(new ServerAddress(host, serverName));
+            }
           }
-        }
-        return this.$q.all(stackPromises);
-      })
-      .then((stacks) => {
-        let stackMap: IMap<string> = <IMap<string>>{};
-        for (let serverGroup in stacks) {
-          stackMap[serverGroup] = stacks[serverGroup];
-        }
-        deferred.resolve(stackMap);
-      });
-    return deferred.promise;
+          return this.$q.all(stackPromises);
+        })
+        .then((stacks) => {
+          let stackMap: IMap<string> = <IMap<string>>{};
+          for (let serverGroup in stacks) {
+            stackMap[serverGroup] = stacks[serverGroup];
+          }
+          deferred.resolve(stackMap);
+        });
+      return deferred.promise;
+    }
   }
 
   getServerGroupCoordinator(serverGroup: IServerGroup): ng.IPromise<IServerAddress> {
     return this.getCoordinatorByServer(serverGroup.members[0], serverGroup.profile);
   }
 
   getCoordinatorByServer(server: IServerAddress, profile: string): ng.IPromise<IServerAddress> {
-    let deferred: ng.IDeferred<IServerAddress> = this.$q.defer<IServerAddress>();
+    if (this.hasJGroupsStack()) {
+      let deferred: ng.IDeferred<IServerAddress> = this.$q.defer<IServerAddress>();
+      this.serverService.getServerStatus(server).then(status => {
+        if (status === "STOPPED") {
+          deferred.reject("It is not possible to connect to server '" + server.toString() + "' as it is stopped");
+        } else {
 
-    this.serverService.getServerStatus(server).then(status => {
-      if (status === "STOPPED") {
-        deferred.reject("It is not possible to connect to server '" + server.toString() + "' as it is stopped");
-      } else {
-
-        this.getChannelNamesByProfile(profile).then(channelNames => {
+          this.getChannelNamesByProfile(profile).then(channelNames => {
             return this.getChannelCoordinator(server, channelNames[0]);
           }).then(view => {
             let coordinator: IServerAddress = this.extractAddressFromView(view);
             deferred.resolve(coordinator);
           });
-      }
-    });
-    return deferred.promise;
+        }
+      });
+      return deferred.promise;
+    } else {
+      this.$q.when(new ServerAddress("N/A", "N/A"));
+    }
   }
 
   getChannelNamesByProfile(profile: string): ng.IPromise<string[]> {
-    let request: IDmrRequest = <IDmrRequest>{
-      address: [].concat("profile", profile, "subsystem", "datagrid-jgroups"),
-      "child-type": "channel"
-    };
-    return this.dmrService.readChildName(request);
+    if (this.hasJGroupsStack()) {
+      let request: IDmrRequest = <IDmrRequest>{
+        address: this.getJgroupsAddress(profile),
+        "child-type": "channel"
+      };
+      return this.dmrService.readChildName(request);
+    } else {
+      this.$q.when(["N/A"]);
```

Comment:
Right, great suggestion! Now hasJGroupsService method invocations are contained almost exclusively to JGroupsService